src/algebra/module/graded_module.lean

```diff
@@ -0,0 +1,136 @@
+/-
+Copyright (c) 2022 Jujian Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jujian Zhang
+-/
+
+import ring_theory.graded_algebra.basic
+
+/-!
+# Graded Module
+
+Given a graded `R`-algebra `A` with grading `𝓐 : ι → submodule R A` and an `A`-module `M` with
+decomposition `ι → addsubmonoid M`,
```

Comment:
We need to be a bit careful here; some references define a graded module as what you'd call a graded module over a graded ring, with `R i * M j <= M (i + j)`.

Reply:
Is that `smul_mem` below